src/server/game/Entities/Unit/Unit.h

```diff
@@ -2297,6 +2305,9 @@ class TC_GAME_API Unit : public WorldObject
         typedef std::list<GameObject*> GameObjectList;
         GameObjectList m_gameObj;
 
+        typedef std::list<AreaTrigger*> AreaTriggerList;
```

Comment:
erasing twice will cause out of range exception :) either .erase and provide an iterator, or std::remove. std::remove will give you a past-end iterator, so if you call .erase with that, it will crash the server.